### DISCUSSION TITLE
Disabilitare entità

### DISCUSSION THREAD

**Discussion by Whitfy** [indomus/forum#1156] [HUB: Home Assistant]
Buongiorno a tutti,
ho una installazione di Home Assistant Core con diversi dispositivi fra cui alcune tapparelle Shelly 2.5 integrate via MQTT. Avrei la necessità di disabilitare l'uso delle tapparelle in alcuni casi, per esempio quando non sono in casa per evitare di far suonare accidentalmente l'allarme.
Per questo sto cercando un modo per realizzarlo e ho pensato per esempio:
- disabilitare un'entità dinamicamente tramite automazione
- modificare dinamicamente il topic o il payload mqtt relativo ai comandi (command_topic)
ovviamente non sono riuscito.
Qualche idea su come potrei fare? Lo scopo è disabilitare i comandi verso la tapparella sia via GUI che via script.

Di seguito un esempio di come ho integrato la tapparella:

```
- platform: mqtt
  name: "Sala da Pranzo"
  unique_id: "6e632100-97ee-4203-a997-6ce67e1ec0ea"
  command_topic: "shellies/SP_TAPP/roller/0/command"
  position_topic: "shellies/SP_TAPP/roller/0/pos"
  state_topic: "shellies/SP_TAPP/roller/0"
  set_position_topic: "shellies/SP_TAPP/roller/0/command/pos"
  availability_topic: "shellies/SP_TAPP/online"
  payload_available: "true"
  payload_not_available: "false"
  retain: false
  payload_open: "open"
  payload_close: "close"
  payload_stop: "stop"
  state_opening: "open"
  state_closing: "close"
  state_stopped: "stop"
  position_open: 100
  position_closed: 0
  optimistic: false
  qos: 1
```

Grazie

**Comment by snakuzzo** [comment]
Non mi pare si possano disabilitare un'entità creata manualmente...come quella MQTT. Puoi farlo di sicuro manualmente con le entità generate dalle integrazioni...anche se in documentazione non trovo come farlo programmaticamente...magari via API.

Per il tuo caso, potresti fare come hai detto, ma non sarebbe banale. Dovresti aggiungere un carattere "sporco" nel `command_topic` e poi ricaricare tutte le entità MQTT create manualmente (via API)
Stesso dicasi per riabilitarle. Pulire il `command_topic` e ricaricare manualmente.

Però dovresti fare uno script (shell?) che vada a cercare e cambiare una stringa (utilizzando `sed` magari) e poi fare una chiamata alle API di HA per ricaricare le enittà manuali

Boh...Sinceramente trovo tutto un po' esagerato. 

Eviterei questa soluzione e magari agirei lato Frontend utilizzando una custom card che si possa nascondere/chiudere con un tap

**Comment by 50494554524F** [comment]
usa ad esempio un input_boolean.away che metterai su on quando esci... o quando lo smartphone non è più connesso alla rete di casa

**Reply by snakuzzo** [reply]
Sì ma lui vuole evitare che, quando è fuori, involontariamente agisca su entità come cover o switch. Con il boolean che funge da "away switch" di fatto, lo useresti per delle condition legate a script ed automazioni. Il controllo delle entità da lovelace funzionerebbe lo stesso.

**Reply by xyzmauro** [reply]
Una conditional card che visualizza le cover solo se il device Tracker é su home?

**Reply by snakuzzo** [reply]
sì...infatti nel thread in alto gli avevo suggerito questa soluzione

**Comment by Whitfy** [comment]
Ok, grazie a tutti, direi che è l'unica soluzione!

Il Sab 11 Dic 2021, 19:43 Diego Ambrosanio ***@***.***> ha
scritto:

**Reply by xyzmauro** [reply]
Giusto per completezza, se vuoi stoppare automazioni o script che interessano le cover quando sei not_home basta che ci metti una condition state home così che se non sei a casa non partono
